### PR TITLE
Add text encoder output caching

### PR DESCRIPTION
**Pull request by RyanJDick**
This PR adds support for text encoder output caching to both the SD v1/v2 and SDXL scripts.

Enabling this option significantly reduces the VRAM requirements. I was able to train an SDXL model on an 8GB GPU with this feature (albeit slowly).

### CODE REVIEW

File: src/invoke_training/training/finetune_lora/finetune_lora_sdxl.py [hunk 1]
```diff
@@ -225,6 +220,43 @@ def _encode_prompt(text_encoders: list[CLIPPreTrainedModel], prompt_token_ids_li
...
+        tokenizer_1 (PreTrainedTokenizer):
+        tokenizer_2 (PreTrainedTokenizer):
+        text_encoder_1 (CLIPPreTrainedModel):
+        text_encoder_2 (CLIPPreTrainedModel):
+    """
+    data_loader = build_image_caption_sdxl_dataloader(
+        config.dataset, tokenizer_1, tokenizer_2, config.train_batch_size, shuffle=False
+    )
+
+    cache = TensorDiskCache(cache_dir)
```
Comment (brandonrising) [inline src/invoke_training/training/finetune_lora/finetune_lora_sdxl.py:244]: how big can this get?

File: src/invoke_training/training/shared/data/transforms/tensor_disk_cache.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import os
+import typing
+
+import torch
+
+
+class TensorDiskCache:
```
Comment (brandonrising) [inline src/invoke_training/training/shared/data/transforms/tensor_disk_cache.py:7]: Do the files on disk ever get removed?

File: src/invoke_training/training/finetune_lora/finetune_lora_sdxl.py (hunk 1 above)
Comment (RyanJDick) [reply src/invoke_training/training/finetune_lora/finetune_lora_sdxl.py:244]: On a test dataset with 833 images, it is ~250MB. So, large enough that it is probably worth cleaning up. I'll add some cleanup logic.

Comment (RyanJDick) [reply src/invoke_training/training/finetune_lora/finetune_lora_sdxl.py:245]: Switched to using a temporary directory in [bf220f9](https://github.com/invoke-ai/invoke-training/pull/17/commits/bf220f9f4a91d343ed41024fdf2d53f44788bc87)

**Pull request merged**